docs/tools/bootloader.md

```diff
@@ -35,12 +38,58 @@ This parameter defines the bootloader image to be used during the built-in postb
+You may use this parameter in conjunction with `target.bootloader_img`, `target.header_format`, `target.header_offset` and `target.app_offset`. It conflicts with `target.mbed_app_start` and `target.mbed_app_size`. When used with `target.bootloader_img`, that parameter defines the start of the application. Otherwise, the start is the start of ROM.
+
+### `target.header_format`
+
+The `target.header_format` configuration key defines an application header as a list of tuples. Each tuple represents a single element of the header format as a name (a valid C identifier), a type, a subtype and finally a single argument. For example, the const type defines subtypes for common sizes of constants, including the 32be subtype that indicates the constant is represented as 32-bit big endian. The following is a list of all types and subtypes that the Mbed OS build tools support.
+
+- `const` A constant value.
+    - `8be` A value represented as 8 bits.
+    - `16be` A value represented as 16 bits big endian.
+    - `32be` A value represented as 32 bits big endian.
+    - `64be` A value represented as 64 bits big endian.
+    - `8le` A value represented as 8 bits.
+    - `16le` A value represented as 16 bits little endian.
+    - `32le` A value represented as 32 bits little endian.
+    - `64le` A value represented as 64 bits little endian.
+
+- `timestamp` A time stamp value in seconds from epoch in the timezone GMT/UTC. The argument to this type is always null.
+    - `64be` A time stamp value truncated to 64 bits big endian.
+    - `64le` A time stamp value truncated to 64 bits little endian.
+
+- `digest` A digest of an image. All digests are computed in the order they appear. A digest of the header digests the header up to its start address.
+    - `CRCITT32be` A big endian 32-bit checksum of an image with an initial value of `0xffffffff`, input reversed, and output reflected.
+    - `CRCITT32le` A little endian 32bit checksum of an image with an initial value of `0xffffffff`, input reversed, and output reflected.
+    - `SHA256` A SHA-2 using a block-size of 256 bits.
+    - `SHA512` A SHA-2 using a block-size of 512 bits.
+
+- `size` The size of a list of images, added together.
+    - `32be` A size represented as 32 bits big endian.
+    - `64be` A size represented as 64 bits big endain.
+    - `32le` A size represented as 32 bits little endian.
+    - `64le` A size represented as 64 bits little endain.
+
+Items in the application header are packed starting where the previous field ended; items in the header are packed using the C "packed" struct semantics. The presence of the `target.header_format` format field defines two macros `MBED_HEADER_START`, which expands to the start address of the firmware header, and `MBED_HEADER_SIZE`, containing the size in bytes of the firmware header.The region following the application header starts at `MBED_HEADER_START + MBED_HEADER_SIZE` rounded up to a multiple of 8 bytes.
```

Comment:
What packs items in the application header?
Note to self: Add missing space after "header."

Reply:
Packed is an adjective. It describes the layout.